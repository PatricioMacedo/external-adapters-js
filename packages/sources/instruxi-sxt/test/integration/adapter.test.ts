import {
  TestAdapter,
  setEnvVariables,
} from '@chainlink/external-adapter-framework/util/testing-utils'
import * as nock from 'nock'
import { mockResponseSuccess, mockResponseError } from './fixtures'
import { endpoint } from '../../src/endpoint/total_reserve'

describe('execute', () => {
  let spy: jest.SpyInstance
  let testAdapter: TestAdapter
  let oldEnv: NodeJS.ProcessEnv

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    process.env.API_ENDPOINT = process.env.API_ENDPOINT ?? 'https://api-endpoint-placeholder.com'
    process.env.API_KEY = process.env.API_KEY ?? 'fake-api-key'
    const mockDate = new Date('2001-01-01T11:11:11.111Z')
    spy = jest.spyOn(Date, 'now').mockReturnValue(mockDate.getTime())

    const adapter = (await import('../../src')).adapter
    adapter.rateLimiting = undefined
    testAdapter = await TestAdapter.startWithMockedCache(adapter, {
      testAdapter: {} as TestAdapter<never>,
    })
  })

  afterAll(async () => {
    setEnvVariables(oldEnv)
    await testAdapter.api.close()
    nock.restore()
    nock.cleanAll()
    spy.mockRestore()
  })

  describe('total_reserve endpoint', () => {
    it('should return success', async () => {
      const data = {
        data: {
          endpoint: 'total_reserve',
          BISCUIT_ATTESTATIONS: 'example_biscuit_attestations_token',
          BISCUIT_BLOCKCHAINS: 'example_biscuit_blockchains_token',
          CHAIN_ID: 'example_chainId',
          ASSET_CONTRACT_ADDRESS: 'example contract address',
          TOKEN_CONTRACT_ADDRESS: 'example token contract address',
          NAMESPACE: 'example_namespace',
        },
      }
      mockResponseSuccess()
      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })

    it('should return error for invalid CHAIN_ID', async () => {
      const data = {
        data: {
          endpoint: 'total_reserve',
          BISCUIT_ATTESTATIONS: 'example_biscuit_attestations_token',
          BISCUIT_BLOCKCHAINS: 'example_biscuit_blockchains_token',
          CHAIN_ID: 'invalid_chain_id',
          ASSET_CONTRACT_ADDRESS: 'example contract address',
          TOKEN_CONTRACT_ADDRESS: 'example token contract address',
          NAMESPACE: 'example_namespace',
        },
      }
      mockResponseError()
      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(400)
      expect(response.json()).toMatchSnapshot()
    })
  })
})
