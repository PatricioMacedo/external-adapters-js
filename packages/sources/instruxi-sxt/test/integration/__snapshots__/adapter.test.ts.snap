// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute total_reserve endpoint should return error for invalid CHAIN_ID 1`] = `
{
  "error": {
    "message": "[Param: BISCUIT_ATTESTATIONS] param is required but no value was provided",
    "name": "AdapterError",
  },
  "status": "errored",
  "statusCode": 400,
}
`;

exports[`execute total_reserve endpoint should return error for missing required parameters 1`] = `
{
  "error": {
    "message": "[Param: BISCUIT_ATTESTATIONS] param is required but no value was provided",
    "name": "AdapterError",
  },
  "status": "errored",
  "statusCode": 400,
}
`;

exports[`execute total_reserve endpoint should return success 1`] = `
{
  "data": {
    "total_reserve": 300000000,
  },
  "result": 300000000,
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
  },
}
`;
